codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/knowledge/GoPointableIndex.java

```diff
@@ -99,8 +96,13 @@ public class GoPointableIndex implements KnowledgeIndex {
             ShapeType.BIG_INTEGER
     );
 
-    // All types types that are comparable to nil
-    private static final Set<ShapeType> NUMBER_SHAPE = SetUtils.of(
+    // Pointer types that are strictly pointer type even if it has required trait
+    private static final Set<ShapeType> STRICT_POINTER_TYPE = SetUtils.of(
+        ShapeType.STRING
+    );
+
+    // All the known pointer type if not required
```

Comment:
Again this only holds True for models we own - but since I've added another Index for `aws` you might be able to keep it.